File: android/app/src/main/java/com/example/autostradaauctions/data/model/Vehicle.kt
package com.example.autostradaauctions.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "vehicles")
data class Vehicle(
    @PrimaryKey
    val id: String,
    val make: String,
    val model: String,
    val year: Int,
    val mileage: Int,
    val engineType: String,
    val transmission: String,
    val fuelType: String,
    val color: String,
    val vin: String,
    val description: String,
    val imageUrls: List<String>,
    val condition: VehicleCondition,
    val isVerified: Boolean = false,
    val createdAt: Long,
    val updatedAt: Long
)

enum class VehicleCondition {
    EXCELLENT,
    GOOD,
    FAIR,
    POOR,
    SALVAGE
}

File: backend/AutostradaAuctions.Api/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace AutostradaAuctions.Api.Models
{
    public class Vehicle
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Make { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Model { get; set; } = string.Empty;

        public int Year { get; set; }

        [StringLength(50)]
        public string Color { get; set; } = string.Empty;

        [Required]
        [StringLength(17)]
        public string VIN { get; set; } = string.Empty;

        public int Mileage { get; set; }

        [StringLength(20)]
        public string FuelType { get; set; } = string.Empty;

        [StringLength(20)]
        public string Transmission { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public List<string> ImageUrls { get; set; } = new List<string>();

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public List<Auction> Auctions { get; set; } = new List<Auction>();
    }
}
